modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/AbstractResourceManager.java

```diff
@@ -64,6 +63,26 @@ public static JSONDecoder getDecoder() throws CharonException {
         return decoder;
     }
 
+    /**
+     * Returns the resource URL builder implementation.
+     *
+     * @return AbstractResourceURLBuilder.
+     */
+    public static AbstractResourceURLBuilder getResourceURLBuilder() {
+
+        return resourceURLBuilder;
+    }
+
+    /**
+     * Sets the resource URL builder implementation.
+     *
+     * @param resourceURLBuilder AbstractResourceURLBuilder.
+     */
+    public static void setResourceURLBuilder(AbstractResourceURLBuilder resourceURLBuilder) {
```

Comment:
Ideally this should be something set at the constructor not a static method.
But, looks like that's the pattern used in this class. Thus, need to check how this class is been used and use the appropriate

Reply:
we cannot define a constructor to this class since it is abstract. Hence kept the static methods.